extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/common/KubernetesStringUtils.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.k8s.overlord.common;
+
+import org.apache.commons.lang3.RegExUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Locale;
+
+public class KubernetesStringUtils
+{
+  public static String parseStringToK8sLabel(String rawString)
+  {
+    return rawString == null ? "" : StringUtils.left(RegExUtils.replaceAll(rawString, "[^a-zA-Z0-9\\\\s]", "")
```

Comment:
I see this is not part of your change, but since we are adding this to a utility function, it would be better to move `"[^a-zA-Z0-9\\\\s]"` to a pattern so that the string does not need to be compiled to a Pattern everytime this function is called.